### ISSUE TITLE
Does not work on Windows

### ISSUE THREAD

**Issue by gamtiq** [tandrewnichols/grunt-simple-npm#1]
The library does not work on Windows (tested on Windows 7). Attempt to run a npm command leads to error: `Fatal error: spawn npm ENOENT`
It seems that the cause is node.js issue joyent/node#2318.
A possible fix is the following:

``` javascript
var name = this.name;
cli.spawn(grunt, this, {}, process.platform === 'win32' ? name + '.cmd' : name);
```

**Comment by tandrewnichols** [comment]
Yeah, I figured I'd see this issue at some point. It was definitely made with Unix in mind, since I only use that. I'll have a look at it and put changes in a branch, but I'll need some help with testing. I'll keep you posted.

**Comment by tandrewnichols** [comment]
Do you know if this is a problem for other commands to? For example, [grunt-simple-git](https://github.com/tandrewnichols/grunt-simple-git). I'm trying to figure out 1) the correct fix. Your proposal seems fine, but I'm still reading through that issue to try to figure out if there's a better way. But also 2) where the fix should go. If this a problem with _every_ `grunt-simple-x` library I've published, then the fix should actually be in [simple-cli](https://github.com/tandrewnichols/simple-cli), where it would work for everything.

**Comment by gamtiq** [comment]
I use grunt-simple-git in a project and all works fine. I suppose this is because Git version for Windows uses `git.exe` as main/target executable file. And as I know the aforementioned node.js issue does not affect `.exe` executables. As a result `spawn` does not have problems with git commands.
So the fix which I proposed is related only to grunt-simple-npm.

**Comment by tandrewnichols** [comment]
Ok, thanks. That makes this much simpler. I'll get a branch up later tonight.

**Comment by tandrewnichols** [comment]
So I changed my mind. Since `simple-cli` is intended to be a wrapper for executing command line binaries via grunt, it should work on windows. It should not be on the consumer of `simple-cli` to implement a windows-compatible solution. I'm opting to replace the default `child_process.spawn` in that library with [win-spawn](https://github.com/ForbesLindesay/win-spawn), which (purportedly) is fully cross-platform. I've got this mostly done, but I'm making a few simplifications to the library as well. I should have a new version of that up in a day or two, and then I can convert this lib to use the new version (that'll be fast). I would guess end-of-week at the latest for a new version of this library.

The new issue is here: tandrewnichols/simple-cli#2

**Comment by gamtiq** [comment]
That seems reasonable.
I agree that fixing simple-cli would be more general and appropriate solution.

**Comment by tandrewnichols** [comment]
I've run into a snag (which I've hit before) involving proxyquire and blanket.js not playing nicely together (both mess with `require.extensions['js']`), so this might take slightly longer than I thought. Sorry for the inconvenience... If you're on a deadline, you could try forking this lib and making your proposed change from above until this is ready to go.

**Comment by gamtiq** [comment]
Actually there is not any inconvenience for me. I have no deadline or hurry.
Anyway thank you for your concern.

**Comment by tandrewnichols** [comment]
Just published v1.0.3 which includes Windows support via simple-cli. Let me know if you have any issues!
